project/ExplorerRay/README.md

```diff
@@ -0,0 +1,87 @@
+# onnxinfo - A tool to show ONNX model summary like torchinfo
+
+This project aims to develop a tool that provides a summary of ONNX models, similar to how torchinfo works for PyTorch models. The tool will focus on offering a layer-wise breakdown, MAC/parameter counts, input/output shapes, and other key model details, all accessible via C++ and Python APIs.
+
+## Basic Information
+
+`onnxinfo` provides a detailed summary of ONNX models, including layer structure and parameter insights, available through both C++ and Python interfaces. 
+
+GitHub URL: https://github.com/ExplorerRay/onnxinfo
+
+## Problem to Solve
+
+With the widespread use of ONNX as an open standard for machine learning models, users used to inspect the structure and parameters of these models by [Netron](https://github.com/lutzroeder/netron). However, it is not that convenient to use it when in a CLI-only environment. This project aims to provide a utility for summarizing ONNX models like torchinfo.
+
+The project applies knowledge from machine learning (deep neural networks) and the ONNX standard, relying on the ONNX Python API for model parsing and analysis.
+
+## Prospective Users
+
+Developers and reasearchers who use ONNX models and need a quick way in python to inspect the model's structure.
+
+## System Architecture
+
+* Input: protocol buffer
+  * ONNX format stores a model graph as a protobuf structure (GraphProto)
+* Process: Iterate nodes in ONNX and parse info, then calculate
+  * ONNX model parsing
+    * Parse ONNX models using the ONNX C++ and Python APIs to get info like layer type (e.g., Conv, Gemm, BatchNorm).
+  * Layer-wise calculating
+    * Input and output shapes
+    * Parameter count (e.g., weights and biases)
+    * MACs (Multiply-Accumulate Operations)
+* Output: Name, Type, Input/Output shapes, MACs, parameters in every layer
```

Comment:
Do you plan to write any data back to the model file(s)?

Reply:
No, just show information about the model.

Reply:
Thanks.  Please make it clear in the proposal.